cpp/src/arrow/scalar.cc

```diff
@@ -194,4 +200,138 @@ Status CheckBufferLength(const FixedSizeBinaryType* t, const std::shared_ptr<Buf
 }
 }  // namespace internal
 
+// CastImpl(...) assumes `to` points to a non null scalar of the correct type with
```

Comment:
Shouldn't all those internal details be pushed in an anonymous namespace?

Reply:
I'll put them in an anonymous namespace